File: C_Programming/Practise_Programs/PP_012.c
//////////////////////////////////////////////////////////////////
//
//  File Name    : Check_Even_Odd
//
//  Description  : This program accepts a number
//                 from the user and checks whether
//                 it is even or odd.
//
//  Example:
//      Input  : 15
//      Output : 15 Number is Odd
//
//////////////////////////////////////////////////////////////////

//////////////////////////////////////////////////////////////////
//
//  Enhanced Version: Added return value function and displays
//                    both the input number and its Even/Odd result.
//
//////////////////////////////////////////////////////////////////

/*
    Algorithm

    START

        Accept number as No

        Call CheckEvenOdd() by passing No

        Receive the result

        If the result is 0
            Display that No is Even

        Otherwise
            Display that No is Odd

    STOP
*/

#include<stdio.h>

int CheckEvenOdd(int iNo)
{
    int iRemainder = 0;

    iRemainder = iNo % 2;

    return iRemainder;
}

int main()
{
    int iValue = 0;
    int iRet = 0;
    
    printf("Enter number to check whether it is Even or Odd : ");
    scanf("%d", &iValue);
   
    iRet = CheckEvenOdd(iValue);

    if(iRet == 0)
    {
        printf("%d Number is Even\n", iValue);
    }
    else
    {
        printf("%d Number is Odd\n", iValue);
    }
    return 0;
}

#include<stdio.h>

int CheckEvenOdd(int iNo)
{
    int iRemainder = 0;

    iRemainder = iNo % 2;

    return iRemainder;
}

int main()
{
    int iValue = 0;
    int iRet = 0;
    
    printf("Enter number to checked whether it is Even or Odd : ");
    scanf("%d", &iValue);
   
    iRet = CheckEvenOdd(iValue);

    if(iRet == 0)
    {
        printf("%d Number is Even\n", iValue);
    }
    else
    {
        printf("%d Number is Odd\n", iValue);
    }
    return 0;
}

File: Java_Programming/Practise_Programs/PP_012.java
//////////////////////////////////////////////////////////////////
//
//  File Name    : PP_012
//
//  Description  : This program accepts a number from the user
//                 and displays all of its factors except the 
//                 number itself using a user-defined class.
//
//  Example:
//      Input  : 12
//      Output : 1
//               2
//               3
//               4
//               6
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class NumberX
{
    public void DisplayFactors(int iNo)
    {
        int iCnt = 0;

        for(iCnt = 1; iCnt < iNo; iCnt++)
        {
            if((iNo % iCnt) == 0)
            {
                System.out.println(iCnt);
            }
        }
    }    
}

class PP_012
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;
        
        System.out.println("Enter number : ");
        iValue = sobj.nextInt(); 
        
        NumberX nobj = new NumberX();

        nobj.DisplayFactors(iValue);
        
    }
}

// Time Complexity : O(N)
// Where N >= 0